### ISSUE TITLE
UploadField is completely inaccessible with a screen reader

### ISSUE THREAD

**Issue by maxime-rainville** [silverstripe/silverstripe-asset-admin#1051]
They are some major accessibility issues with the UploadField that will make it unusual for a user relying on a screen reader:
* The label for the field point to a non-existent ID
* The Upload field doesn't have a `aria-role` or any other aria attribute
* Once a file is selected, the "View" and "Deselect" buttons don't have any aria label to communicate their purpose

@silverstripeux have many other misgivings with the way [their original design](https://projects.invisionapp.com/dsm/silver-stripe/silver-stripe/asset/components/5b5e6e3984472000117f3ee4) has been implemented. These issues are worth addressing in an upcoming minor release.

However, because the UploadField is unusable for a class of users with no alternative, I think it's worth fixing the accessibility issues retroactively with patch releases for older minor releases.
